### DISCUSSION TITLE
Telecamere IMOU e rilevamento movimento

### DISCUSSION THREAD

**Discussion by array81** [indomus/forum#1749] [HUB: Home Assistant]
Ho 3 telecamere della IMOU (2 modello Cue 2C e una modello Cue2), da quanto mi risulta tutte compatibili Onvif profilo S tanto da usarle tramite Onvif su un NVR. Riesco integrare tutte e 3 le telecamere in Home Assistant ma a parte il visualizzare il video e riavviarle non riesco a fare altro. Pensavo di poter avere un trigger al rilevamento del movimento o del suono, cosa che posso fare sia sia con l'app ufficiale che con il mio NVR cinese. 
Qualche consiglio in merito su come riuscire a fargli importare queste entità?

**Comment by avvalex** [comment]
La parola magica da cercare in Google è "ONVIF2MQTT"

**Comment by array81** [comment]
Ho cercato e trovato il progetto che sembra molto interessante.
Peccato che a quanto pare non funzioni su Raspberry. L'autore sembra abbia compilato il software solo per amd64 e non per arm64, quindi quando provo ad installarlo di fatto non funziona.

**Reply by avvalex** [reply]
Io ho tutto in Docker, l'immagine compilata per Raspberry è "freakern/onvif2mqtt".

**Comment by array81** [comment]
Intanto grazie per il suggerimento.
Ho installato l'immagine dopo aver preparato il file di configurazione.
L'idea è buona ma non sembra funzionare a dovere, almeno con le mie 3 camere IMOU. Spesso il movimento non viene rilevato sebbene ci sia e altrettanto spesso ho eventi OFF quando dovrei avere eventi ON.

Ho trovato questo: https://github.com/berfenger/cam2mqtt
Volevo provarlo ma dopo aver installati l'immagine e preparato il file di configurazione quando provo ad avviare l'immagine con il seguente comando:
`docker run -v /home/array81/cam2mqtt:/config acasal/cam2mqtt`
ottengo un messaggio di errore, mi pare di capire che non trova il file di configurazione eppure il file config.yml è nella cartella indicata.
Non sono un esperto, qualcuno può suggerirmi come risolvere?

**Comment by avvalex** [comment]
Che modello sono le IMOU?

**Comment by array81** [comment]
Ho una Cue 2 e due Cue 2C

**Comment by avvalex** [comment]
Cone la mia Cue 2 sembra funzionare a dovere...

**Comment by array81** [comment]
Ho fatto qualche test con MQTT Explorer e sono giunto alle seguenti conclusioni.
Credo che gli eventi ON e OFF funzionassero nel seguente modo, ON veniva inviato al rilevamento del movimento OFF invece quando il movimento cessava. Invece non è così. ON è inviato al rilevamento del movimento ma a questo seguono poi una serie di eventi OFF anche se il movimento c'è ancora. Tra l'altro il primo OFF quasi sempre arriva quasi subito dopo l'ON. Immagino sia il protocollo ONVIF ad essere in questo modo. Fatto sta che effettivamente si può catturare l'evento NO MOVIMENTO -> SI MOVIMENTO ma nulla altro, non si può sapere quanto il movimento dura e se in un determinato momento c'è o no dl movimento.

**Comment by avvalex** [comment]
Effettivamente a me serve solo sapere quando qualcosa si muove. Credo che la cam verifichi quando c'è movimento e poi mandi messaggio, che poi resta valido tipo 10/15 secondi, poi verifica ancora, ma intanto magari parte un off.
Io creerei un binary sensore che diventa off dopo tot secondi che non c'è un movimento.

**Comment by lozanca** [comment]
> Ho 3 telecamere della IMOU (2 modello Cue 2C e una modello Cue2), da quanto mi risulta tutte compatibili Onvif profilo S tanto da usarle tramite Onvif su un NVR. Riesco integrare tutte e 3 le telecamere in Home Assistant ma a parte il visualizzare il video e riavviarle non riesco a fare altro. Pensavo di poter avere un trigger al rilevamento del movimento o del suono, cosa che posso fare sia sia con l'app ufficiale che con il mio NVR cinese. Qualche consiglio in merito su come riuscire a fargli importare queste entità?

Ciao ti chiedo come hai fatto ad integrare 3 telecamere. io ho 3 C22C e 1 F22D. Riesco ad integrare solo 1 C22C e una F22D. se aggiungo al seconda C22C vanno in conflitto e devo eliminarla. Per fare questo uso Onvif. Mi date una mano? grazie

**Reply by array81** [reply]
Non ci sono riuscito, da quanto ho capito è un banale errore di configurazione della telecamere. Il problema mi pare si verifichi se uno aggiunge una seconda telecamera IMOU di uno stesso modello di uno già inserito. A quel punto invece di aggiungere una nuova telecamera viene modificata quella esistente cambiando la password di quella esistente con quella nuova. Il risultato è che la nuova non viene aggiunta e quella esistente smette di funzionare. Sul forum di HA ci sono altri utenti con lo stesso problema e hanno tutti telecamere IMOU.

**Comment by lozanca** [comment]
grazie per la risposta... quindi non è possibile aggiungere piu telecamere
uguali. non ci sono trucchi per ovviare al problema?

Il giorno mer 28 set 2022 alle ore 17:42 Martin ***@***.***>
ha scritto:

**Reply by avvalex** [reply]
A dire il vero io ho due CUE 2 che funzionano bene, integrate normalmente.
Mi vengono in mente 2 possibilità:
1) integrarle manualmente tramite flusso RSTP;
2) Scaricare SmartPSS dal sito Dahua e, nella configurazione della Cam, in Device Config, System, General, cambiare il nome della telecamera....

**Reply by lozanca** [reply]
provato ma mi HA sovrascrive le cam..... non so piu che fare

**Comment by lozanca** [comment]
Grazie per la info.
Provo a fare come hai detto con le 2 due soluzioni.
Una domanda: ma le tue due CUE 2 quindi non vanno in conflitto?

grazie ancora

Il giorno gio 29 set 2022 alle ore 14:36 avvalex ***@***.***>
ha scritto:

**Reply by avvalex** [reply]
Assolutamente no.

**Comment by array81** [comment]
Io ci ho rinunciato sinceramente ne ho provate molte ma non ci sono riuscito (anche il cambio di nome). Alla fine mi sono limitato ad usare Onvif2Mqtt come suggeritomi per avere il trigger del movimento.

**Comment by array81** [comment]
Ho la soluzione al problema. Non è merito mio, l'ho trovata sul forum ufficiale di HA.
Da quanto ho capito il problema sta nel fatto che alcune telecamere hanno 2 indirizzi MAC, uno usato quando si connette direttamente il telefono alla telecamera (al setup iniziale) mentre l'altro usato normalmente quando si connette la telecamera alla rete.
Mentre il secondo MAC è univoco e legato all'hardware il primo in alcuni modelli di telecamere (soprattutto IMOU) è uguale.
Ora quando si aggiunge una nuova telecamera su HA usa l'indirizzo MAC come unique_id e nel caso in questione usa il primo indirizzo (quello comune) e non il secondo. Quando si prova ad aggiungere la seconda telecamera HA rileva che c'è già un dispositivo con lo stesso unique_id e quindi pensa che si tratti dello stesso dispositivo.
Al momento sembra che l'unica soluzione sia la seguente:

1. installare la prima telecamera normalmente;
2. aprire il file /.storage/core.config_entries che si trova all'interno della cartella di HA (quella dove si trova il file YAML per la configurazione). Attenzione si tratta di una cartella invisibile;
3. trovate la sezione del codice relativa alla vostra telecamera (basta cercare il nome che le avete assegnato);
4. modificare il codice assegnato a unique_id;
5. riavviare HA;
6. installare normalmente la seconda telecamera;

Se avete altre telecamere dello stesso modello la cosa andrà ripetuta tutte le volte. Ricordate di riavviare HA dopo aver modificato il file.
Ho testato personalmente la cosa e funziona.

**Reply by ortorob73** [reply]
Ciao

ho provato la procedura con le mie telecamere esterne, stesso modello, ma la procedura fallisce.

Alternative? Suggerimenti?

Grazie

**Reply by array81** [reply]
No, io da allora ho le 3 telecamere IMOU funzionanti e non ho dovuto fare altro. Mi dispiace.

**Comment by lozanca** [comment]
Grazie mille....
finalmente ho tutte le 6 camere in HA...
Aiuto preziosissimo.

Il giorno ven 7 ott 2022 alle ore 15:09 Martin ***@***.***>
ha scritto:

**Comment by achiosi** [comment]
Ciao @array81 , posso chiederti qual e' l'NVR cinesi che usi con le telecamere IMOU (cue2 e cue2c)?
Grazie!

**Reply by array81** [reply]
Non ha una marca. In sostanza c'è una società cinese di cui ora sinceramente non ricordo il nome che produce schede per NVR, poi i singoli produttori prendono queste schede e le mettono in box rimarchiati. Si riconoscono perchè funzionano tutti con l'app per smartphone xmeye. Non sono il massimo della privacy tanto che io ho bloccato all'NVR l'accesso a internet tramite router. Ma per avere un registratore locale a 15/20 euro vanno bene.

**Reply by achiosi** [reply]
Grazie mille per l'info! Da quanto ho capito il supporto ONVIF delle IMOU non e' eccellente per cui e' utile sapere che c'e' un NVR per che puo' usare il motion detection come trigger.

**Reply by array81** [reply]
Si questi funzionano anche se dopo un aggiornamento ha smesso di registrarmi l'audio. Considera però che il mio è un modello vecchio e non rilasciano più aggiornamenti, magari hanno risolto. Il movimento viene rilevato ed è possibile registrare solo quando qualcosa si muove.

**Reply by Baldo1981** [reply]
Hi, I also have a Chinese NVR.  I have 3 onvif IMOUE ipcams.  The NVR detects them, but by entering the password of the respective cameras I can never view the video in the NVR.  What can I do?  A thousand thanks

**Comment by Danybs72** [comment]
chiedo umilmente scusa sono un neofita assoluto e probabilmente troverete particolamente elementare la mia domanda. Ho comperato 2 imou Cue 2C per videosorvegliare casa con l'unica pretesa di rilevare eventuali effrazioni. Scaricata l'app Imou lite ho notato che entrambe le cam funzionavano regolarmente. Ma quest'app , a parer mio poco intuitiva, non mi dà la possibilità di poter mettere in pausa la registrazione ? Ovvero a me 1 : non interessa essere rilevato con relativo messaggio sullo smartphone ogni qualvolta giro per casa , 2 ma esiste un modo d'impostare la notifica sul telefono (tipo una sirena) in modo che prescinda dalle altre notifiche ? Ho provato a "giocarci " un pochino poi ho desistito. Ringrazio chiunque voglia aiutarmi scusandomi a priori perchè magari è una cosa assolutamente banale da risolvere,
